### DISCUSSION TITLE
Plugins: possible to reset the state of the document?

### DISCUSSION THREAD

**Discussion by tobiasBora** [xournalpp/xournalpp#6099] [Q&A]
I need to create a plugin that heavily modifies a document (it will create many new pages, change layer visibility as motivated in https://github.com/xournalpp/xournalpp/issues/6073, or even create new strokes and move them to create animations etc) just to export it to a PDF file, but then it will need to revert everything after the production of the PDF file. Is there a simple way to revert the document to a state at time X, without keeping track of all changes manually? For instance, is it possible to get the whole document structure, with all strokes/layers/pages/etc, save it, and revert it later?

**Comment by rolandlo** [comment]
You could save the document (as a xopp-file) and reload it, but that's about what you can do.

**Reply by tobiasBora** [reply]
Thanks… That might be better than nothing, but is this possible **without** having a user interaction/dialog? I can see `fileDialogSave` but it seems like the user will always be prompted for something…
